src/main/java/edu/ksu/canvas/attendance/controller/CourseConfigurationController.java

```diff
@@ -93,6 +116,44 @@ public ModelAndView synchronizeWithCanvas(@PathVariable String sectionId) throws
         return page;
     }
 
+    @RequestMapping(value = "/{sectionId}/save", params = "pushGradesToCanvas", method = RequestMethod.POST)
+    public ModelAndView pushGradesToCanvas(@PathVariable String sectionId, @ModelAttribute("courseConfigurationForm") @Valid CourseConfigurationForm classSetupForm, BindingResult bindingResult) throws NoLtiSessionException{
+        LOG.info("eid: " + canvasService.getEid() + " is pushing grades for course # " + canvasService.getCourseId() + " to Canvas");
+        ModelAndView page = new ModelAndView("forward:/courseConfiguration/" + sectionId);
+
+        Long courseId = Long.valueOf(canvasService.getCourseId());
+
+        boolean isSimpleAttendance = classSetupForm.getSimpleAttendance();
+        List<AttendanceSummaryModel> summaryForSections = isSimpleAttendance ?
+                reportService.getSimpleAttendanceSummaryReport(Long.parseLong(sectionId)) :
+                reportService.getAviationAttendanceSummaryReport(Long.parseLong(sectionId));
+
+        List<Error> submissionErrors = assignmentSubmitter.submitCourseAttendances(isSimpleAttendance, summaryForSections, courseId, canvasService.getOauthToken(), classSetupForm);
+
+        if(submissionErrors != null && !submissionErrors.isEmpty()){
+            List<String> errors = new ArrayList<>();
+            submissionErrors.forEach(error -> errors.add(error.getMessage()));
+            page.addObject("error" , errors);
+        } else {
+            page.addObject("pushingSuccessful", true);
+        }
+        return page;
+    }
+
+    @RequestMapping(value = "/{sectionId}/delete", method = RequestMethod.POST)
```

Comment:
I'm conflicted about using POST here. There is a DELETE method defined in HTTP. However, that usually implies you are deleting a resource on the server. In this case you are deleting a resource in canvas.